backend/src/ml_space_lambda/utils/common_functions.py

```diff
@@ -119,6 +136,12 @@ def wrapper(event, context):
         logger.info(f"Lambda {lambda_func_name}({code_func_name}) invoked with {_sanitize_event(event)}")
         try:
             result = f(event, context)
+            if isinstance(result, ApiResponse):
+                return {
+                    "statusCode": result.status_code,
+                    "headers": result.headers,
+                    "body": result.body
+                }
```

Comment:
100% up to you but I'd recommend implementing this using inversion of control so the ApiResponse can render itself. It shouldn't matter much with such a limited response object possible for use with APIGW but it'll also clean up the code a little. It would also allow you to have more explicit variants like JsonApiResponse or StringApiResponse instead of hiding special handling based types in the implementation details like is happening on line 65.

```python
            if isinstance(result, ApiResponse):
                 return result.render() // or whatever you want to call it
```